src/ngircd/irc-channel.c

```diff
@@ -264,8 +313,27 @@ join_send_topic(CLIENT *Client, CLIENT *target, CHANNEL *chan,
 	if (Client_Type(Client) != CLIENT_USER)
 		return true;
 	/* acknowledge join */
+	#ifndef IMPLEMENT_EXTENDED_JOIN
 	if (!IRC_WriteStrClientPrefix(Client, target, "JOIN :%s", channame))
 		return false;
+	#endif
+	#ifdef IMPLEMENT_EXTENDED_JOIN
+	CONN_ID conn;
+	conn = Client_Conn(Client);
+	if(Client_Cap(Client) & CLIENT_CAP_EXTENDED_JOIN) {
+		char * account_name;
+		if(Client_AccountName(Client)) {
+			account_name = Client_AccountName(Client);
+		}
+		else {
+			account_name = "*";
+		}
+		Conn_WriteStr(conn, ":%s JOIN %s %s :%s", Client_MaskCloaked(Client), channame, account_name, Client_Info(Client));
+		}
+		else {
```

Comment:
Indentation is broken here (the `else` should be one level up), which makes it a bit hard to read.
And please remove the unnecessary braces (`{`, `}`) here as well – see my comment above.